src/umpire/Allocator.inl

```diff
@@ -28,7 +29,13 @@ inline void* Allocator::allocate(std::size_t bytes)
   if (0 == bytes) {
     ret = allocateNull();
   } else {
-    ret = m_allocator->allocate(bytes);
+    try {
```

Comment:
Does the `try` itself add any additional overhead to this call?  I would hope that it is just the `catch` that has all of the cost.  Should performance be checked?